azuread/resource_application_test.go

```diff
@@ -203,6 +202,37 @@ func TestAccAzureADApplication_native(t *testing.T) {
 	})
 }
 
+func TestAccAzureADApplication_nativeReplyUrls(t *testing.T) {
+	resourceName := "azuread_application.test"
+	id := uuid.New().String()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckADApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccADApplication_nativeReplyUrls(id),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckADApplicationExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", fmt.Sprintf("acctest%s", id)),
+					resource.TestCheckResourceAttr(resourceName, "type", "native"),
+					resource.TestCheckResourceAttr(resourceName, "reply_urls.#", "1"),
+					// I didn't understand the reason reply_urls.3714513888 was used in TestAccAzureADApplication_update
```

Comment:
when the type is a set u need to get the set "hash value". I believe this can be grabbed from the state/plan when running locally and then put in here as it will always be the same for a given value.